main_test.go

```diff
@@ -147,3 +232,37 @@ func readFile(dir, file string) string {
 	data, _ := ioutil.ReadFile(filename)
 	return string(data)
 }
+
+// getTags returns all of the tags in a git repository as a map.
+func getTags(dir string) (map[string]bool, error) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+	if err := os.Chdir(dir); err != nil {
+		return nil, err
+	}
+	cmd := exec.Command("git", "tag")
```

Comment:
Just a minor note ... you can do `cmd.Dir=dir` instead of `os.Chdir`, which will change the working directory just for that command. Or if you have to do `os.Chdir` I recommend doing this at the top of the function:

```diff
cwd, err := os.Getwd()
+defer os.Chdir(cwd)
```

This will make sure the working directory is always reset, just in case we exit early due to error. Not a big deal at all, just thought you might be interested in some feedback :)